frontend/src/composables/use-external-sources.ts

```diff
@@ -0,0 +1,40 @@
+import { computed } from "vue"
+
+import { useFeatureFlagStore } from "~/stores/feature-flag"
+import { useSearchStore } from "~/stores/search"
+import {
+  IMAGE,
+  isAdditionalSearchType,
+  isSupportedMediaType,
+} from "~/constants/media"
+import { getAdditionalSources } from "~/utils/get-additional-sources"
+
+export const useExternalSources = () => {
+  /**
+   * External sources search form shows the external sources for current search type.
+   */
+  const externalSourcesType = computed(() => {
+    const featureFlagStore = useFeatureFlagStore()
+    const searchStore = useSearchStore()
+    const searchType = searchStore.searchType
+    if (
+      isSupportedMediaType(searchType) ||
+      (featureFlagStore.isOn("additional_search_types") &&
+        isAdditionalSearchType(searchType))
+    ) {
+      return searchType
+    }
+    return IMAGE
+  })
+  const externalSources = computed(() => {
+    const searchStore = useSearchStore()
+    const query = searchStore.searchQueryParams
+    const type = externalSourcesType.value
+    return getAdditionalSources(type, query)
+  })
```

Comment:
Is it safe to call `useSearchStore` etc inside the computed call-backs?

Reply:
My reasons for putting the `useStore` inside the computed were incorrect:
You cannot call `useStore` from outside of a `setup` function of a component because then `pinia` instance is not available. However, since we only call this composable from a component's `setup` function, `useStore` can be placed inside the composable. (The problem would arise if we put the `useStore` outside the `useExternalSources` function, or if we called `useExternalSources` from a middleware, for instance).

I moved the `useStore` calls to the composable itself, and added a comment on calling the composable from a `setup` function.

Reply:
Just for my knowledge, should we avoid naming things that _aren't_ called during `setup` `use*` and composables? Would that be a useful disambiguator?

Also, _is_ it safe to call them inside a computed callback? I genuinely don't know and can't remember the precise conditions that cause a memory leak :slightly_smiling_face:. I remember it had to do with allocating a new ref in a callback, but I don't know if store callbacks, etc, cause that issue?

One more thing, would [something like this ESLint plugin (maybe not exactly that one as it's new and small)](https://github.com/ktsn/eslint-plugin-vue-composable) be useful to prevent some cases of mistakes? Apparently the main [Vue plugin for ESLint decided not to implement any composable rules](https://github.com/vuejs/eslint-plugin-vue/issues/2093), but only because of the naming convention issue.

Reply:
> Also, is it safe to call them inside a computed callback? I genuinely don't know and can't remember the precise conditions that cause a memory leak 🙂. I remember it had to do with allocating a new ref in a callback, but I don't know if store callbacks, etc, cause that issue?
I also do not remember. I'll try to look it back to have a better idea (and document, if possible). An ESLint plugin sounds great!